### ISSUE TITLE
Feature request: add support of national colors for roads references in instruction display for car

### ISSUE THREAD

**Issue by Lispach** [osmandapp/OsmAnd#7786]
Hello, there.

I thought about something that could improve text instructions clarity: in a given country, it is frequent that a given road type will have its `ref` displayed with specific colours on the traffic signs. For instance, in France, the _routes nationales_, whose `ref` necessarily starts with a `N`, always have their `ref` displayed, on traffic signs, with a white font on a red background. The colors are directly linked to the `ref`: a `ref` starting with `N` will always be written white upon red. The drivers are acustomed to the color coding system, so it would be great that the text instructions displayed on the top of the screen during navigation also displayed it. Instead of displaying, for instance, _N 85 » Paris_, the display would be like _![image](https://user-images.githubusercontent.com/56482311/67794686-5b3b9c00-fa7d-11e9-8a2f-b8c8b3e3b119.png) » Paris_, to display the `ref` as it is displayed on site and, therefore, make the text instruction clearer.

The implementation of this feature could be done by a config file containing a list of countries, say listed by ISO-3166 codes, and, for each given country, could allow contributors to add the colour coding system of their country, for instance with a list of regex on the `ref`. For example, for France, the config file would look like this:
```
FR => {
    "N.*" => {
        background => "red";
        text => "white";
    }
}
```
So, when a user in France would use car navigation, and the text instruction contains a `ref` which matches the regex "N.*", the `ref` in the text instruction would be displayed with the specified colours. These colours could be given in RGB, to allow country-specific colours.

As a fallback rule, in the case the text instruction contains a `ref` for which, in the current country, no rule exists or none has a regex matching the `ref`, the display would be the current one. In addition, the regex matching would stop on the first matching regex, to prevent a potential bug if several regex matches a given `ref`.

I hope I expressed my idea in an understandable way. Feel free to ask me if a clarification is needed.

Regards.

**Comment by OpenSourceForEver** [comment]
The idea is very good but no configuration file is needed. In OSM there is a colour:ref tag.

**Comment by vshcherb** [comment]
Yes we have national shields for each country, probably we need to double check France

**Comment by xmd5a2** [comment]
@vshcherb this is request for UI, not for backend. These shields are already rendered at the map.

**Comment by Lispach** [comment]
It seems that the map shields already use the national shield color systems. If I'm right about this, you could simply replace the reference, in the instruction text displayed, by its shield. Note that, in this case, I think you should drop the comma after the ref in the instruction text, as a shield followed by a comma would be a bit strange.

**Comment by Davileci** [comment]
E 70 & E 05 signals are well rendered, but GI-2132 & GI-2134 are not.

![IMG_20240217_195848_771.jpg](https://github.com/osmandapp/OsmAnd/assets/97534508/8583225a-43db-446a-b077-e65b129332de)

The number of characters (6) of the ref impacts the rendering of the rectangular board with green or orange colours.

![Screenshot_20230918_024643_OsmAnd~.jpg](https://github.com/osmandapp/OsmAnd/assets/97534508/5f534e42-40c4-474c-a027-5045cad2f20f)

**Comment by vshcherb** [comment]
@xmd5a2 We can fix the issues in the previous comment to do
![image](https://github.com/osmandapp/OsmAnd/assets/1042025/e8f3d2d4-cb32-49b2-a576-e081a5bdde3c)
